openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -445,165 +323,208 @@ public static Builder builder() {
+         * Returns the Jandex {@link IndexView} containing annotated endpoint
+         * classes.
          *
-         * @param className name of the model reader class
-         * @return updated builder instance
+         * @return {@code IndexView} containing endpoint classes
          */
-        public Builder modelReader(String className) {
-            Objects.requireNonNull(className, "modelReader class name must be non-null");
-            apiConfigBuilder.modelReader(className);
-            return this;
-        }
+        public abstract IndexView indexView();
 
         /**
-         * Set the app-provided OpenAPI model filter class.
+         * Makes sure the set-up for OpenAPI is consistent, internally and with
+         * the current Helidon runtime environment (SE or MP).
          *
-         * @param className name of the filter class
-         * @return updated builder instance
+         * @throws IllegalStateException if validation fails
          */
-        public Builder filter(String className) {
-            Objects.requireNonNull(className, "filter class name must be non-null");
-            apiConfigBuilder.filter(className);
-            return this;
+        public void validate() throws IllegalStateException {
         }
 
         /**
-         * Sets the servers which offer the endpoints in the OpenAPI document.
+         * Path under which to register OpenAPI endpoint on the web server.
          *
-         * @param serverList comma-separated list of servers
+         * @param path webContext to use, defaults to
+         * {@value DEFAULT_WEB_CONTEXT}
          * @return updated builder instance
          */
-        public Builder servers(String serverList) {
-            Objects.requireNonNull(serverList, "serverList must be non-null");
-            apiConfigBuilder.servers(serverList);
+        public Builder webContext(String path) {
+            this.webContext = Optional.of(path);
             return this;
         }
 
         /**
-         * Adds an operation server for a given operation ID.
+         * Sets the location of the static OpenAPI document file.
          *
-         * @param operationID operation ID to which the server corresponds
-         * @param operationServer name of the server to add for this operation
+         * @param path non-null location of the static OpenAPI document file
          * @return updated builder instance
          */
-        public Builder addOperationServer(String operationID, String operationServer) {
-            Objects.requireNonNull(operationID, "operationID must be non-null");
-            Objects.requireNonNull(operationServer, "operationServer must be non-null");
-            apiConfigBuilder.addOperationServer(operationID, operationServer);
+        public Builder staticFile(String path) {
+            Objects.requireNonNull(path, "path to static file must be non-null");
+            staticFilePath = Optional.of(path);
```

Comment:
Using an `Optional` here seems purposeless.

Reply:
The code uses the `Optional` aspect basically as a flag, whether an assignment was made to the static file path. If so, then `staticfile()` uses the assigned path; if not, it looks for a default path. We could have used a null value instead to indicate the lack of an explicit assignment.